src/flow/ext_multiroute_flow.jl

```diff
@@ -0,0 +1,230 @@
+function emrf{T<:AbstractFloat,R<:Number}(
+  flow_graph::DiGraph,                   # the input graph
+  source::Int,                           # the source vertex
+  target::Int,                           # the target vertex
+  capacity_matrix::AbstractArray{T,2},   # edge flow capacities
+  flow_algorithm::AbstractFlowAlgorithm, # keyword argument for algorithm
+  routes::R = 0
+  )
+  breakingpoints = breakingPoints(flow_graph,source,target,capacity_matrix)
+  if routes > zero(R)
+    x,f = intersection(breakingpoints,routes)
+    return maximum_flow(flow_graph,source,target,capacity_matrix,
+      algorithm=flow_algorithm,restriction=x)
+  end
+  return breakingpoints
+end
+
+"""
+Output a set of (point,slope) that compose the restricted max-flow function.
+One point by possible slope is enough (hence O(λ*max_flow) complexity).
+Requires arguments:
+- flow_graph::DiGraph                    # the input graph
+- source::Int                            # the source vertex
+- target::Int                            # the target vertex
+- capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+"""
+
+function auxiliaryPoints{T<:AbstractFloat}(
+  flow_graph::DiGraph,                   # the input graph
+  source::Int,                           # the source vertex
+  target::Int,                           # the target vertex
+  capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+  )
+  # Problem descriptors
+  λ = maximum_flow(flow_graph,source,target)[1] # Connectivity
+  n = nv(flow_graph) # number of nodes
+  r1, r2 = minmaxCapacity(capacity_matrix) # restriction left (1) and right (2)
+  auxpoints = fill((0.,0.),λ+1)
+
+  # Initialisation of left side (1)
+  f1, F1, cut1 = maximum_flow(flow_graph,source,target,capacity_matrix,
+                        algorithm=BoykovKolmogorovAlgorithm(),restriction=r1)
+  s1 = slope(flow_graph,capacity_matrix,cut1,r1) # left slope
+  auxpoints[λ + 1 - s1] = (r1,f1) # Add left initial auxiliary point
+
+  # Initialisation of right side (2)
+  f2, F2, cut2 = maximum_flow(flow_graph,source,target,capacity_matrix,
+                        algorithm=BoykovKolmogorovAlgorithm(),restriction=r2)
+  s2 = slope(flow_graph,capacity_matrix,cut2,r2) # right slope
+  auxpoints[λ + 1 - s2] = (r2,f2) # Add right initial auxiliary point
+
+  # Loop if the slopes are distinct by at least 2
+  if s1 > s2 + 1
+    queue = [((f1,s1,r1),(f2,s2,r2))]
+
+    while !isempty(queue)
+      # Computes an intersection (middle) with a new associated slope
+      (f1,s1,r1),(f2,s2,r2) = pop!(queue)
+      r, expectedflow = intersection(r1,f1,s1,r2,f2,s2)
+      f, F, cut = maximum_flow(flow_graph,source,target,capacity_matrix,
+                        algorithm=BoykovKolmogorovAlgorithm(),restriction=r)
+      s = slope(flow_graph,capacity_matrix,max(cut,1),r) # current slope
+      auxpoints[λ + 1 - s] = (r,f)
+      # If the flow at the intersection (middle) is as expected
+      if expectedflow ≈ f
```

Comment:
It is actually ≈ (\approx), to avoid problems (infinite loop) because of floating numbers precision.

I will add a comment, for instance:
`if expectedflow ≈ f # approximatively equal (enforced by floating precision)`